### ISSUE TITLE
SecurityError: The operation is insecure.

### ISSUE THREAD

**Issue by KK-MarcelJahn** [Modernizr/Modernizr#1825]
Hi,

we are currently using a custom Build (v. 2.8.3) and also tried the same build based on 3.2 but we are still receiving a SecurityError in Firefox.

This Error is thrown when using Firefox without cookies. So whenever we block Cookies within Firefox the whole application crashes due to this error.

As far as I know Firefox is pretty strict when cookies are blocked. So for example access to localStorage or sessionStorage will return in a SecurityError. So maybe this is the reason for the error we are experiencing....

This is the only detail I can see is:

SecurityError: The operation is insecure.
...{},T.on=function(e,t){this._l[e]||(this._l[e]=[]),this._l[e].push(t),Modernizr.h...

I hope someone can help us.

**Comment by patrickkettner** [comment]
Could you share the build you are using?

**Comment by KK-MarcelJahn** [comment]
Thanks for the quick response!

We are using this build:
http://modernizr.com/download/#-fontface-backgroundsize-borderimage-borderradius-boxshadow-flexbox-hsla-multiplebgs-opacity-rgba-textshadow-cssanimations-csscolumns-generatedcontent-cssgradients-cssreflections-csstransforms-csstransforms3d-csstransitions-applicationcache-canvas-canvastext-draganddrop-hashchange-history-audio-video-indexeddb-input-inputtypes-localstorage-postmessage-sessionstorage-websockets-websqldatabase-webworkers-geolocation-inlinesvg-smil-svg-svgclippaths-touch-webgl-shiv-mq-cssclasses-addtest-prefixed-teststyles-testprop-testallprops-hasevent-prefixes-domprefixes-svg_filters-load

Thank you!

**Comment by KK-MarcelJahn** [comment]
Hi,

After a bit of digging i found that at least the following tests causing issues:

localStorage,
sessionStorage,
indexedDB

I'll remove them from my build to get a stable system, but it would be great if someone could come up with a safe test.

**Comment by patrickkettner** [comment]
is this an error or a console warning?

**Comment by KK-MarcelJahn** [comment]
Unfortunately it's an error and the script is not being executed which is causing a lot of issues.
After removing LocalStorage, SessionStorage and indexedDB everything is working again.

You may want to read this:
http://meyerweb.com/eric/thoughts/2012/04/25/firefox-failing-localstorage/

It's about cookie expiration but it's the same when blocking cookies.

**Comment by patrickkettner** [comment]
I disabled cookies in firefox and could not replicate any of the issues you mentioned. Could you provide a demo jsbin or codepen or something that shows the problem, as well as the specific version of firefox you are using?

thanks!

**Comment by KK-MarcelJahn** [comment]
Hi Patrick, 

I couldn't replicate the issue on osx. Our customers are using Firefox 43 on Windows (7 & 8).
So i had to use browserstack to reproduce this error. 

Firefox on osx doesn't seem to be as strict as on Windows.

**Comment by patrickkettner** [comment]
Hi @KK-MarcelJahn!
I am still not able to replicate. I am testing a build with localStorage, sessionStorage, and indexedDB on Firefox 43 on windows 10. 

Can you please provide a link to a demo jsbin or codepen that shows the problem and include exact steps needed to replicate the problem when starting from a stock firefox install

thanks!

**Comment by sheff146** [comment]
It really occurs. We faced this issue in Firefox 43 both on Windows and OS X. Modernizr 2.8.3 installed from bower. SecurityError on IndexedDB test.

**Comment by KK-MarcelJahn** [comment]
Hi @patrickkettner,

simply go to http://html5doctor.com/demos/modernizr/canvas-yes.htm, block cookies and reload.
It was working fine in FF 42 on OSX, but after updating to FF 43 i now also see the error on my mac. (as @sheff146 mentioned) 

![screenshot 2016-01-12 09 34 25](https://cloud.githubusercontent.com/assets/3899649/12258538/b9acd08c-b90f-11e5-9331-ecf34eb5c0c8.png)

Thank you!

**Comment by patrickkettner** [comment]
have you tried Modernizr 3.3.0? Modernizr 2.X is not longer supported. I am
not able to replicate any issue on Modernizr 3

On Tue, Jan 12, 2016 at 12:35 AM, Marcel Jahn <notifications@github.com>
wrote:

> Hi @patrickkettner <https://github.com/patrickkettner>,
>
> simply go to http://html5doctor.com/demos/modernizr/canvas-yes.htm, block
> cookies and reload.
> It was working fine in FF 42 on OSX, but after updating to FF 43 i now
> also see the error on my mac. (as @sheff146 <https://github.com/sheff146>
> mentioned)
>
> [image: screenshot 2016-01-12 09 34 25]
> <https://cloud.githubusercontent.com/assets/3899649/12258538/b9acd08c-b90f-11e5-9331-ecf34eb5c0c8.png>
>
> Thank you!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Modernizr/Modernizr/issues/1825#issuecomment-170837580>
> .
>



-- 
patrick

**Comment by KK-MarcelJahn** [comment]
Unfortunately v3.3.0 isn't working as well. It throws the same error as it's accessing localStorage, sessionStorage and indexedDb without further checking or 'protection'.

http://codepen.io/anon/pen/jWwKvG

![screenshot 2016-01-12 11 54 38](https://cloud.githubusercontent.com/assets/3899649/12261734/4d2be1dc-b923-11e5-9003-585a072b012d.png)

**Comment by ryanseddon** [comment]
Ok I've manage to reproduce your error you werent clear on how you disabled cookies but I found the ui in your screenshot.

<img width="396" alt="site-info" src="https://cloud.githubusercontent.com/assets/143402/12279378/d348bdd2-b9da-11e5-9853-1597a0aab7d9.png">

<img width="388" alt="more-info" src="https://cloud.githubusercontent.com/assets/143402/12279383/dd9d168e-b9da-11e5-8bb0-0577ad79e2ee.png">

<img width="712" alt="disable-cookies" src="https://cloud.githubusercontent.com/assets/143402/12279384/e18b4c2a-b9da-11e5-8682-b67c13446573.png">

Basically when trying to access any storage property when cookies are disabled throws a security issue, local/sessionStorage don't throw as we have try catch error handling for other issues, in indexDB we just need to wrap the [check](https://github.com/Modernizr/Modernizr/blob/master/feature-detects/indexeddb.js#L19) in a try catch.

@KK-MarcelJahn wanna do a PR to fix this?

**Comment by sheff146** [comment]
Hello guys! 

I've submitted a PR https://github.com/Modernizr/Modernizr/pull/1831 for this issue. Could you take a look?

Thanks.

**Comment by patrickkettner** [comment]
fixed with #1831

**Comment by SonaliAvaya** [comment]
![Still facing "Security" error ]
(https://user-images.githubusercontent.com/62263280/83305533-fa20a600-a1ce-11ea-81cf-04bb0cf450cb.png)

**Comment by rejas** [comment]
@SonaliAvaya I am not sure this is the place to post this. Sicne it seems you got this error from running cypress. Or where are you using modernizr?

**Comment by SonaliAvaya** [comment]
No i am running specs file on cypress
and cypress supports firefox 67 beta version which i am running

**Comment by rejas** [comment]
Again I ask @SonaliAvaya : are you using modernizr in your project?

**Comment by SonaliAvaya** [comment]
No i am not using modernizr but i am using cypress and having issue

**Comment by patrickkettner** [comment]
@rejas the original comment appears to be deleted, so I think we are good.

@SonaliAvaya as I am sure you noted, this is not the correct place for the issue. Best of luck solving it!
